ui/tests/unit/services/permissions-test.js

```diff
@@ -148,77 +93,161 @@ module('Unit | Service | permissions', function (hooks) {
+
+      this.service.set('exactPaths', getPaths());
+      assert.true(
+        this.service.hasNavPermission('access', ['methods', 'mfa', 'oidc']),
+        'hasNavPermission returns true for array of route params when any route is permitted'
+      );
+      assert.false(
+        this.service.hasNavPermission('access', ['methods', 'mfa', 'oidc'], true),
+        'hasNavPermission returns false for array of route params when any route is not permitted and requireAll is passed'
+      );
+
+      this.service.set('exactPaths', getPaths('read'));
+      assert.true(
+        this.service.hasNavPermission('access', ['methods', 'mfa', 'oidc'], true),
+        'hasNavPermission returns true for array of route params when all routes are permitted and requireAll is passed'
+      );
+
+      this.service.set('exactPaths', getPaths('deny'));
+      assert.false(
+        this.service.hasNavPermission('access', ['methods', 'mfa', 'oidc']),
+        'hasNavPermission returns false for array of route params when no routes are permitted'
+      );
+      assert.false(
+        this.service.hasNavPermission('access', ['methods', 'mfa', 'oidc'], true),
+        'hasNavPermission returns false for array of route params when no routes are permitted and requireAll is passed'
+      );
+    });
   });
 
-  test('appends the namespace to the path if there is one', function (assert) {
-    const namespaceService = Service.extend({
-      path: 'marketing',
+  module('pathWithNamespace', function () {
+    test('appends the namespace to the path if there is one', function (assert) {
+      const namespaceService = Service.extend({
+        path: 'marketing',
+      });
+      this.owner.register('service:namespace', namespaceService);
+      assert.strictEqual(this.service.pathNameWithNamespace('sys/auth'), 'marketing/sys/auth');
+    });
+
+    test('appends the chroot and namespace when both present', function (assert) {
+      const namespaceService = Service.extend({
+        path: 'marketing',
+      });
+      this.owner.register('service:namespace', namespaceService);
+      this.service.set('chrootNamespace', 'admin/');
+      assert.strictEqual(this.service.pathNameWithNamespace('sys/auth'), 'admin/marketing/sys/auth');
+    });
+    test('appends the chroot when no namespace', function (assert) {
+      this.service.set('chrootNamespace', 'admin');
+      assert.strictEqual(this.service.pathNameWithNamespace('sys/auth'), 'admin/sys/auth');
+    });
+    test('handles superfluous slashes', function (assert) {
+      const namespaceService = Service.extend({
+        path: '/marketing',
+      });
+      this.owner.register('service:namespace', namespaceService);
+      this.service.set('chrootNamespace', '/admin/');
+      assert.strictEqual(this.service.pathNameWithNamespace('/sys/auth'), 'admin/marketing/sys/auth');
+      assert.strictEqual(
+        this.service.pathNameWithNamespace('/sys/policies/'),
+        'admin/marketing/sys/policies/'
```

Comment:
is this expected to have the path end in a `/`?

Reply:
Yes, for glob paths they end in `/`